### ISSUE TITLE
Potentially autograding code placement

### ISSUE THREAD

**Issue by JaredMiller2022** [jamesjuett/examma-ray#65]
I'm not sure how the autograding functionality works under the hood for the coding questions, so if the issue I describe below is actually just a result of us not creating thorough enough test cases, then feel free to disregard this issue...

When grading list-flatten, I sometimes found that the autograde functionality would give points for having both the (node == nullptr) base case and the (node->next == nullptr) test case even when the latter came before the former. However, this would be incorrect code because if (node == nullptr), then the code would error on the access to node->next. It would be nice if in cases similar to this, we could have the AG check to make sure the (node == nullptr) base case occurs before the (node->next == nullptr) base case.

Note: when grading, I made sure to check for this error & filled everybody else in on it too, so this didn't actually affect grades.

**Comment by jamesjuett** [comment]
Good catch. Yeah, this is a current limitation of the semantic analysis toolset in Lobster, that there isn't an implementation of checking for sequencing.
